warehouse-management-service/pkg/database/database.go

```diff
@@ -0,0 +1,5 @@
+package database
+
+type Service interface {
```

Comment:
minor: My suggestion for moving this on the consumer side still stands, and in addition to that we should call this interface [`Closer`](https://go.dev/doc/effective_go#interface-names)

Reply:
The intent of the Service interface was to compose functions that would be used by the application to work with the database. This only declares `Close()` for now, but would be extended to include functions to perform CRUD operations.

But I do understand why this is wrongly placed and should live on the client side and the database package should only provide concrete implementations.